when building python for qemux86-64 on ubuntu 11.10/64bit
it gropes into host includes and then mixes them with cross
includes and as a result some modules fail to compile and link
one of the modules is python-elementtree which is then not
found during image creation

Proble is that setup.py tries to add native includes that newer
ubuntu has introduced for multiarch support. But that should
only happen for native builds and not cross building python
so we add a check here.

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Pending

###############################################################################
# Modified to work with meta-debian
# Checking CROSS_COMPILE should be at the begin of function
# to prevent all cases of adding native include/lib to compiler flag
###############################################################################


diff --git a/setup.py b/setup.py
index ba2d242..bf859be 100644
--- a/setup.py
+++ b/setup.py
@@ -391,6 +391,8 @@ class PyBuildExt(build_ext):
              self.failed.append(ext.name)
 
     def add_multiarch_paths(self):
+        if os.environ.get('CROSS_COMPILE') is not None:
+            return
         # Debian/Ubuntu multiarch support.
         # https://wiki.ubuntu.com/MultiarchSpec
         cc = sysconfig.get_config_var('CC')
